### PR TITLE
fix(useOverflowContainer): Removes double overflowManager creation

### PR DESCRIPTION
**Pull request by ling1726**
Adds an effect dependency exception to overflowManager creation in `useOverflowContainer` as it resulted in always recreating the overflowManager because the first mount flag would change.

### CODE REVIEW

File: packages/react-components/react-overflow/stories/src/Overflow/Default.stories.tsx [hunk 1]
```diff
@@ -1,22 +1,42 @@
 import * as React from 'react';
+
```
Comment (miroslavstastny) [inline packages/react-components/react-overflow/stories/src/Overflow/Default.stories.tsx:2]: I guess this is accidentally pushed?

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.099 MB`<br />`271.641 kB` | `1.099 MB`<br />`271.641 kB` | `-2 B` <br />|
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr> | `12.81 kB`<br />`4.821 kB` | `12.808 kB`<br />`4.819 kB` | `-2 B` <br />`-2 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-breadcrumb</samp> <br /> <abbr title='bundle-size/index.fixture.js'>@fluentui/react-breadcrumb - package</abbr> | `113.699 kB`<br />`31.498 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.14 kB`<br />`20.137 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `218.197 kB`<br />`63.203 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.447 kB`<br />`14.59 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `107.387 kB`<br />`35.758 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/416d7ddb3bd0ef5187c151807f10a0cfa721e0c1'>416d7ddb3bd0ef5187c151807f10a0cfa721e0c1</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-overflow/stories/src/Overflow/Default.stories.tsx (hunk 1 above)
Comment (ling1726) [reply packages/react-components/react-overflow/stories/src/Overflow/Default.stories.tsx]: ah good shout, thanks

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Avatar-mount_base.html">617</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Avatar-mount.html">623</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Button-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Button-mount.html">301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Field-mount_base.html">1099</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Field-mount.html">1153</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProvider-mount_base.html">727</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProvider-mount.html">726</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProviderWithTheme-mount_base.html">84</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProviderWithTheme-mount.html">87</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">31</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/MakeStyles-mount_base.html">841</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/MakeStyles-mount.html">884</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Persona-mount_base.html">1803</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/Persona-mount.html">1715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/SpinButton-mount_base.html">1418</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/SpinButton-mount.html">1386</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/SwatchPicker-mount_base.html">1708</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32459/perf-test-react-components/SwatchPicker-mount.html">1638</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-overflow-f9979cd2-b20b-4857-97e8-4e9436f5353d.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-overflow-f9979cd2-b20b-4857-97e8-4e9436f5353d.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32459&commitId=4bf82cbc03343804b5df1ad0f520b1a2067ca926&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|34| Changed|
|Avatar Converged.badgeMask.chromium.png|36| Changed|

</details>


  <details><summary><b>Drawer</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.overlay drawer full.chromium.png|3315| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by miroslavstastny — APPROVED** [review]

**Pull request merged**